gen/ioctl/generate.sh

```diff
@@ -44,5 +44,8 @@ qemu-riscv64 -L /usr/riscv64-linux-gnu ./main.exe >> "$out"
 s390x-linux-gnu-gcc -Iinclude -c list.c $cflags
 s390x-linux-gnu-gcc main.c list.o -o main.exe $cflags
 qemu-s390x -L /usr/s390x-linux-gnu ./main.exe >> "$out"
+loongarch64-linux-gnu-gcc -Iinclude -c list.c $cflags
+loongarch64-linux-gnu-gcc main.c list.o -o main.exe $cflags
+./main.exe >> "$out"
```

Comment:
Hmm this is implying the host environment can transparently run LoongArch binaries. See how other non-x86 arches launch their artifacts with qemu linux-user emulation.

But, as we don't have Debian/Ubuntu cross toolchains available yet, in fact even `loongarch64-linux-gnu-gcc` is not supposed to be available. How are we going to deal with this?

Reply:
qemu linux-user emulation method use the "-L /usr/xxx-linux-gnu",which specifies the library path to be searched by the qemu simulator. This path is usually set by the cross-compilation toolchain installer. 
So no matter which method you use,  compile the cross tool chain is the first step, 
Maybe use the qemu simulator is more complicated, since most os use a lower 
qemu version which does not support loongarch yet

Reply:
Yeah, the key point is: this script would be run by any contributor to this library, and adding those changes as-is would affect anyone not familiar with LoongArch development, which I suppose is the majority. (Also not everyone uses `loongarch64-linux-gnu`; Gentoo for example uses `loongarch64-unknown-linux-gnu`, notice the vendor field.)

We may have to disable the part for everyone else for the time being: maybe wrap them in a `if false; then ... fi` block with some explanations attached.

Reply:
The vendor in the triple is configured by itself,  eg, centos/fedora use x86_64-redhat-linux-gcc, alpine use x86_64-alpine-linux-musl-gcc, so use the common case without unknown

Reply:
> The vendor in the triple is configured by itself, eg, centos/fedora use x86_64-redhat-linux-gcc, alpine use x86_64-alpine-linux-musl-gcc, so use the common case without unknown

I'm not saying that we "should" choose one triple over another, but rather there's simply *not* a common case right now. Other arches are apparently using Debian cross tuples, but we don't currently have such a cross toolchain package available upstream, so `loongarch64-linux-gnu` is equally unusable as other variants so far.

Reply:
https://wiki.debian.org/Ports/loong64 has publised the loongarch64 target triple "loongarch64-linux-gnu"
So we can use the name as most arches do. yes, We can't get such a cross toolchain directly from the upstream now, 
we can compile it.  And debian community work is also in progress.

Reply:
So for now, would it work to just check in pregenerated loongarch64 files as separate files in the repo, and just have this gen/ioctl/generate.sh script just `cat` them into the output? That way, it doesn't depend on a toolchain that ubuntu-latest users won't have an easy way to install.

Reply:
Sorry, I don't fully understand what you mean. Do you mean that our current patch only adds these files in the output src directory, or just temporarily removes the part of the gen/ioctl/generate.sh file that uses the loongarch cross-tool chain until the upstream cross-tool chain is available?

Reply:
@zhaixiaojuan I believe it's just a matter of moving the LoongArch part of the generated output to another file, so we can temporarily avoid calling into the toolchain non-existent on others' systems.

In other words, instead of:

```sh
loongarch64-linux-gnu-gcc -Iinclude -c list.c $cflags
loongarch64-linux-gnu-gcc main.c list.o -o main.exe $cflags
./main.exe >> "$out"
```

You should instead do this for the time being:

```sh
# cross LoongArch toolchain is not yet packaged in mainstream distros yet,
# so pre-generated output is used for the time being
cat loongarch-ioctls.txt >> "$out"
```

Reply:
thanks